resources/party.ts

```diff
@@ -21,7 +21,7 @@ export default class PartyTracker {
   allianceIds_: string[] = [];
   nameToRole_: { [name: string]: Role } = {};
   idToName_: { [id: string]: string } = {};
-  roleToPartyNames_: Record<Role, string[]> = emptyRoleToPartyNames();
```

Comment:
I think `Record` is fine for cases where the key value is a union type.  It's not that I want to remove all uses of `Record`.  It's more that I think `Record<string, etc>` and `Record<number, etc>` are clearer as index types?